test/libsolidity/util/TestFileParser.cpp

```diff
@@ -0,0 +1,301 @@
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <test/libsolidity/util/TestFileParser.h>
+
+#include <test/Options.h>
+#include <boost/algorithm/string.hpp>
+#include <boost/algorithm/string/predicate.hpp>
+#include <boost/optional.hpp>
+#include <boost/throw_exception.hpp>
+#include <fstream>
+#include <memory>
+#include <stdexcept>
+
+using namespace dev;
+using namespace langutil;
+using namespace solidity;
+using namespace dev::solidity::test;
+using namespace std;
+
+namespace
+{
+	void expect(string::iterator& _it, string::iterator _end, string::value_type _c)
+	{
+		if (_it == _end || *_it != _c)
+			throw Error(Error::Type::ParserError, string("Invalid test expectation. Expected: \"") + _c + "\".");
+		++_it;
+	}
+
+	template<typename IteratorType>
+	void skipWhitespace(IteratorType& _it, IteratorType _end)
+	{
+		while (_it != _end && isspace(*_it))
+			++_it;
+	}
+
+	template<typename IteratorType>
+	void skipSlashes(IteratorType& _it, IteratorType _end)
+	{
+		while (_it != _end && *_it == '/')
+			++_it;
+	}
+}
+
+pair<bytes, ByteFormats> TestFileParser::formattedStringToBytes(string _string)
+{
+	bytes result;
+	vector<ByteFormat> formats;
+	auto it = _string.begin();
+	while (it != _string.end())
+	{
+		if (isdigit(*it) || (*it == '-' && (it + 1) != _string.end() && isdigit(*(it + 1))))
+		{
+			ByteFormat format;
+			format.type = ByteFormat::UnsignedDec;
+
+			bool isNegative = (*it == '-');
+			if (isNegative)
+				format.type = ByteFormat::SignedDec;
+
+			auto valueBegin = it;
+			while (it != _string.end() && !isspace(*it) && *it != ',')
+				++it;
+
+			bytes newBytes;
+			try {
+				u256 numberValue{string{valueBegin, it}};
+				// TODO: Convert to compact big endian if padded
+				if (numberValue == u256{0})
+					newBytes = bytes{0};
+				else
+					newBytes = toBigEndian(numberValue);
+				result += newBytes;
+				formats.push_back(std::move(format));
+			}
+			catch (std::exception const&)
+			{
+				throw Error(Error::Type::ParserError, "Argument encoding invalid.");
+			}
+		}
+		else
+			throw Error(Error::Type::ParserError, "Argument encoding invalid.");
+
+		skipWhitespace(it, _string.end());
+		if (it != _string.end())
+			expect(it, _string.end(), ',');
+		skipWhitespace(it, _string.end());
+	}
+	return make_pair(result, formats);
+}
+
+string TestFileParser::bytesToFormattedString(bytes const& _bytes, ByteFormats const& _formats)
+{
+	// TODO: Convert from compact big endian if padded
+	auto it = _bytes.begin();
+	stringstream resultStream;
+	for (auto const& format: _formats)
+	{
+		bytes byteRange{it, it + format.size};
+		switch(format.type)
+		{
+			case ByteFormat::SignedDec:
+				if (*byteRange.begin() & 0x80)
+				{
+					for (auto& v: byteRange)
+						v ^= 0xFF;
```

Comment:
I wonder whether it would be better to perform this transform on the u256 (there is a helper function called `u2s`).